compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -1078,38 +1071,40 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                             }
                         }
                     } else {
-                        err.span_label(
-                            fn_call_span,
-                            &format!(
-                                "{} {}moved due to this method call{}",
-                                place_name, partially_str, loop_message
-                            ),
-                        );
+                        let label = CaptureCausedBy::MethodCall {
+                            place_name: &place_name,
+                            partially_str,
+                            loop_message,
+                            span: fn_call_span,
+                        };
+                        err.subdiagnostic(label);
                     }
                     // Avoid pointing to the same function in multiple different
                     // error messages.
                     if span != DUMMY_SP && self.fn_self_span_reported.insert(self_arg.span) {
-                        err.span_note(
-                            self_arg.span,
-                            &format!("this function takes ownership of the receiver `self`, which moves {}", place_name)
-                        );
+                        let label = CaptureCausedBy::FnTakeSelf {
+                            place_name: &place_name,
+                            span: self_arg.span,
+                        };
+                        err.subdiagnostic(label);
```

Comment:
```suggestion
                        err.subdiagnostic(CaptureCausedBy::FnTakeSelf {
                            place_name: &place_name,
                            span: self_arg.span,
                        });
```
similarly elsewhere, no need to add the extra binding